logstash-core/spec/logstash/pipeline_spec.rb

```diff
@@ -276,6 +276,36 @@ class TestPipeline < LogStash::Pipeline
     end
   end
 
+  describe "max inflight warning" do
+    let(:config) { "input { dummyinput {} } output { dummyoutput {} }" }
+    let(:batch_size) { 1 }
+    let(:pipeline) { LogStash::Pipeline.new(config, :pipeline_batch_size => batch_size, :pipeline_workers => 1) }
+    let(:logger) { pipeline.logger }
+    let(:warning_prefix) { /CAUTION: Recommended inflight events max exceeded!/ }
+
+    before(:each) do
+      allow(LogStash::Plugin).to receive(:lookup).with("input", "dummyinput").and_return(DummyInput)
+      allow(LogStash::Plugin).to receive(:lookup).with("codec", "plain").and_return(DummyCodec)
+      allow(LogStash::Plugin).to receive(:lookup).with("output", "dummyoutput").and_return(DummyOutput)
+      allow(logger).to receive(:warn)
+      thread = Thread.new { pipeline.run }
```

Comment:
This validation is actually done in start_workers. It needs to be there to avoid calling `safe_pipeline_worker_count` twice.

That should be refactored at some point, but I don't think that needs to happen here.

https://github.com/elastic/logstash/pull/4391/files should fix that issue a little bit, and ultimately removing the multiline filter should fully fix it (we won't have to do the check).